doc/man3/CMS_get0_type.pod

```diff
@@ -15,20 +15,26 @@ CMS_get0_type, CMS_set1_eContentType, CMS_get0_eContentType, CMS_get0_content -
 
 =head1 DESCRIPTION
 
-CMS_get0_type() returns the content type of a CMS_ContentInfo structure as
-and ASN1_OBJECT pointer. An application can then decide how to process the
+The CMS_get0_type() function returns a pointer to the content type of a CMS_ContentInfo structure as
+an ASN1_OBJECT pointer.  This is typically set when the CMS_ContentInfo structure is created, or when a signed or encrypted artifact is decoded.
+An application can then decide how to process the
 CMS_ContentInfo structure based on this value.
 
 CMS_set1_eContentType() sets the embedded content type of a CMS_ContentInfo
-structure. It should be called with CMS functions with the B<CMS_PARTIAL>
+structure. It should be called with CMS functions (such as L<CMS_sign>, L<CMS_encrypt>)
+with the B<CMS_PARTIAL>
 flag and B<before> the structure is finalised, otherwise the results are
 undefined.
 
 ASN1_OBJECT *CMS_get0_eContentType() returns a pointer to the embedded
-content type.
+content type.  It differs from CMS_get0_type in that it points to the actual
+eContentType element contained in the ASN1 structure that was constructed or received.
```

Comment:
Somehow this description wasn't quite enough for me to understand what it did. I still had to check the source. Perhaps an example would help?